backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/cfg/ElasticsearchBackendSettings.java

```diff
@@ -300,6 +300,23 @@ private ElasticsearchBackendSettings() {
 	 */
 	public static final String MAX_KEEP_ALIVE = "max_keep_alive";
 
+	/**
+	 * This property defines how many shards <strong>must fail</strong> for response to be considered
+	 * a failure when there's at least one successful shard.
+	 * <p>
+	 * In case all shards fail, Elasticsearch cluster will return a 400 status code itself,
+	 * but if only some of the shards fail, then the client will receive a successful partial response from the shards
+	 * that were successful.
+	 * <p>
+	 * To prevent getting any partial results this setting can be set to {@code 0}.
+	 * <p>
+	 * Expects a zero or positive Integer value,
+	 * or a String that can be parsed into such Integer value.
+	 * <p>
+	 * Defaults to {@link Defaults#MAX_ALLOWED_SHARD_FAILURES}.
+	 */
+	public static final String MAX_ALLOWED_SHARD_FAILURES = "max_allowed_shard_failures";
```

Comment:
> I can't really imagine someone setting this to anything other than 0 pr Integer.MAX_VALUE.

true 😃. I've started with a simple boolean flag, but then looking at the response, I thought that maybe someone could find it useful to get at least some partial results. So I wanted to discuss that with you 😃. I think that if we go with allowing partial failures, then we probably would also need to have that somehow exposed to the user, too (maybe something like `SearchResult#isPartialResult()` or something along those lines...

Reply:
> I think that if we go with allowing partial failures, then we probably would also need to have that somehow exposed to the user, too (maybe something like SearchResult#isPartialResult() or something along those lines...

Right. And that doesn't seem like something we should do in this bugfix that we want to backport.

Reply:
Oh, and I just realized that the `search` prefix isn't great as we'd get `hibernate.search.search.shard_failure.ignore`. Maybe `hibernate.search.query.shard_failure.ignore`? Since we already use the "query" prefix in the ORM mapper and Lucene backend.